test/tests_for_sdkmr/EE2Runjob_test.py

```diff
@@ -0,0 +1,268 @@
+    ji.service_ver = "git5678"
+    ji.source_ws_objects = ["1/2/3", "4/5/6"]
+    ji.parent_job_id = "None"
+    jr = JobRequirements()
+    jr.clientgroup = "grotesquememlong"
+    jr.cpu = "4"
+    jr.memory = "3"
+    jr.disk = "26"
+    ji.requirements = jr
+    ji.narrative_cell_info = Meta()
+    expected_job.job_input = ji
+    assert len(sdkmr.save_job.call_args_list) == 2
+    got_job = sdkmr.save_job.call_args_list[0][0][0]
+    assert_jobs_equal(got_job, expected_job)
+
+    kafka.send_kafka_message.assert_any_call(
+        KafkaCreateJob("someuser", "603051cfaf2e3401b0500982")
+    )
+    condor.run_job.assert_called_once_with(
+        params={
+            "method": "lolcats.lol_unto_death",
+            "source_ws_objects": ["1/2/3", "4/5/6"],
+            "service_ver": "git5678",
+            "job_id": "603051cfaf2e3401b0500982",
+            "user_id": "someuser",
+            "token": "tokentokentoken",
+            "cg_resources_requirements": {
+                "client_group": "grotesquememlong",
+                "request_cpus": "4",
+                "request_memory": "32",
+            },
+        },
+        concierge_params=None,
+    )
+
+    # updated job data save
+    mongo.get_job.assert_called_once_with("603051cfaf2e3401b0500982")
+
+    # update to queued state
+    got_job = sdkmr.save_job.call_args_list[1][0][0]
+    expected_job = Job()
+    expected_job.id = ObjectId("603051cfaf2e3401b0500982")
+    expected_job.status = "queued"
+    # no way to test this really without code refactoring
+    expected_job.queued = got_job.queued
+
+    expected_job.scheduler_type = "condor"
+    expected_job.scheduler_id = "cluster42"
+    assert_jobs_equal(got_job, expected_job)
+
+    kafka.send_kafka_message.assert_called_with(  # update to queued state
+        KafkaQueueChange(
+            job_id="603051cfaf2e3401b0500982",
+            new_status="queued",
+            previous_status="created",
+            scheduler_id="cluster42",
+        )
+    )
+    slack.run_job_message.assert_called_once_with(
+        "603051cfaf2e3401b0500982", "cluster42", "someuser"
+    )
+
+
+def assert_jobs_equal(got_job: Job, expected_job: Job):
+    """
+    Checks that the two jobs are equivalent, except that the 'updated' fields are checked that
+    they're within 1 second of each other.
+    """
+    # I could not get assert got_job == expected_job to ever work, or even
+    # JobRequirements1 == JobRequirements2 to work when their fields were identical, so we
+    # do this disgusting hack instead. Note it will need to be updated any time a job field is
+    # added.
+    if not hasattr(got_job, "id"):
+        assert not hasattr(expected_job, "id")
+    else:
+        assert got_job.id == expected_job.id
+    assert got_job.user == expected_job.user
+    assert got_job.authstrat == expected_job.authstrat
+    assert got_job.wsid == expected_job.wsid
+    assert got_job.status == expected_job.status
```

Comment:
all these repetitive statements are giving me refactoritis! Suggestions for refactoring if you can't do a straight object equality:
- make an explicit list of all the properties and do
```py
for property in property_list:
  assert got_job[property] == expected_job[property]
```
- list the keys of expected_job and do the same thing

It should be easy to omit `updated` from the check, whatever you choose to do.

Reply:
Hmm, I'd have to use `getattr` or go into the instance dict for that, which I'm a bit meh about as it's a bit magic-y. How painful is the refactoritis?

Reply:
what's wrong with `getattr` or the instance dict? They are pretty standard methods.

Reply:
I'm not so sure they are. Directly accessing the instance dict is very rare in my experience, and IMO A Very Bad Thing In General. 

`getattr` is more common, but still pretty rare in my experience. But fine, I'll to it with `getattr`. Yeccch.

Reply:
MongoEngine does some weird stuff with the `__dict__`:

```
    print(repr(got_job))
    print(vars(got_job))
    print(got_job.__dict__)
    print(getattr(got_job, 'updated'))
```
results in
```
{"user": "someuser", "authstrat": "kbaseworkspace", "status": "created", "updated": 1614038718.449384, "job_input": {"method": "lolcats.lol_unto_death", "service_ver": "git5678", "source_ws_objects": ["1/2/3", "4/5/6"], "parent_job_id": "None", "requirements": {"clientgroup": "grotesquememlong", "cpu": 4, "memory": 32, "disk": 2600}, "narrative_cell_info": {}}, "child_jobs": [], "batch_job": false}
{'_cls': 'Job'}
{'_cls': 'Job'}
1614038718.449384
{'_cls': 'Job'}
```

Reply:
done

Reply:
It is possible to get a dict like so

job.to_mongo().to_dict() from both job objects and then compare the dictionaries